src/kademlia/node.cpp

```diff
@@ -1250,17 +1087,13 @@ void node::incoming_request(msg const& m, entry& e)
 				TORRENT_ASSERT(sizeof(to_add.sig) == msg_keys[4].string_length());
 				memcpy(to_add.value, buf.first, buf.second);
 				memcpy(&to_add.key, pk, sizeof(to_add.key));
-		
-				boost::tie(i, boost::tuples::ignore) = m_mutable_table.insert(
-					std::make_pair(target, to_add));
-				m_counters.inc_stats_counter(counters::dht_mutable_data);
 
-//				fprintf(stderr, "added mutable item (%d)\n", int(m_mutable_table.size()));
+				m_storage.put_mutable_item(target, to_add, m.addr.address());
 			}
 			else
 			{
 				// this is the case where we already 
-				dht_mutable_item* item = &i->second;
+				dht_mutable_item* item = (dht_mutable_item*)i;
```

Comment:
why the cast? I would expect this to cause a compile error on travis (because of -Werror)